internal/fileutils/fileutils.go

```diff
@@ -39,3 +39,32 @@ func Touch(path string) error {
 	}
 	return file.Close()
 }
+
+// Copy copies a file from a source to a destination path, preserving the file mode.
+func Copy(srcPath, destPath string) error {
+	src, err := os.Open(srcPath)
+	if err != nil {
+		return err
+	}
+	defer src.Close()
+
+	dst, err := os.Create(destPath)
```

Comment:
We can move the `os.Stat(srcPath)` here and then use `os.OpenFile(dst, O_RDWR|O_CREATE|O_TRUNC, info.Mode())` to create the file instead of using `Chmod` (so one syscall less).

Reply:
Neat!